spras/diamond.py

```diff
@@ -0,0 +1,120 @@
+        seeds_df = seeds_df.sort_values(by=[Dataset.NODE_ID], ascending=True, ignore_index=True)
+        seeds_df.to_csv(filename_map['seeds'], index=False, columns=[Dataset.NODE_ID], header=None)
+
+        # Create network file
+        edges_df = data.get_interactome()
+        edges_df = convert_directed_to_undirected(edges_df)
+        edges_df.to_csv(filename_map["network"], columns=["Interactor1", "Interactor2"], index=False, header=None, sep=',')
+
+    @staticmethod
+    def run(seeds=None, network=None, output_file=None, n=200, alpha=1, container_framework="docker"):
+        """
+        Run DIAMOnD with Docker
+        @param seeds: input seeds (required)
+        @param network: input network file (required)
+        @param container_framework: choose the container runtime framework, currently supports "docker" or "singularity" (optional)
+        @param n: the desired number of DIAMOnD genes to add.
+        @param alpha: int representing weight of the seeds (default: 1)
+        @param output_file: path to the output pathway file (required)
+        """
+        if not seeds or not network or not output_file:
+            raise ValueError('DIAMOnD arguments are missing')
+
+        # TODO we should dynamically calculate n.
+
+        work_dir = '/apsp'
+
+        # Each volume is a tuple (src, dest)
+        volumes = list()
+
+        bind_path, seeds_file = prepare_volume(seeds, work_dir)
+        volumes.append(bind_path)
+
+        bind_path, network_file = prepare_volume(network, work_dir)
+        volumes.append(bind_path)
+
+        # Create the parent directories for the output file if needed
+        Path(output_file).parent.mkdir(parents=True, exist_ok=True)
+        bind_path, mapped_out_file = prepare_volume(output_file, work_dir)
+        volumes.append(bind_path)
+
+        command = ['python',
+                   '/DIAMOnD.py',
+                   network_file,
+                   seeds_file,
+                   str(n),
+                   str(alpha),
+                   mapped_out_file]
+
+        print('Running DIAMOnD with arguments: {}'.format(' '.join(command)), flush=True)
+
+        container_suffix = "diamond:latest"
+        out = run_container(
+                            container_framework,
+                            container_suffix,
+                            command,
+                            volumes,
+                            work_dir)
+        print(out)
+
+    @staticmethod
+    def parse_output(raw_pathway_file, standardized_pathway_file, original_dataset: Dataset):
+        df = raw_pathway_df(raw_pathway_file, sep='\t', header=0)
+        if not df.empty:
+            # preprocessing - drop [useful] p_hyper information, and rename columns to Rank and Node
+            df = df.drop(columns=["p_hyper"])
+            df.columns = ["Rank", "Node"]
+
+            # TODO: we throw away rank information - we want this.
```

Comment:
Actually, how do we want to handle this? Take the edges with vertex rankings 1-4 and 2-3. Should we have this as 'decorative vertex data' (suggested in #88) or do we want some user input about what ranking algorithm to use on edges given vertex rankings?

Reply:
Perhaps, since we do have `Rank 1` constant ranks, we can have these ranks to be the same (edge rank = vertex 1 rank + vertex 2 rank), and add the original vertex ranks as decorative vertex data from #88.